api/application.go

```diff
@@ -993,7 +993,13 @@ func (h ApplicationHandler) AssessmentList(ctx *gin.Context) {
 	resources := []Assessment{}
 	for i := range assessments {
 		r := Assessment{}
-		r.With(&assessments[i])
```

Comment:
There's already a questionnaire resolver instantiated in this method that you can reuse (line 970), you don't need to instantiate a new one. These assessment-related resolver objects cache the results of their operations, so it's best to reuse them when possible and especially avoid instantiating them in a loop unless its necessary.

Reply:
Also, if you were to add `"Assessments.Questionnaire"` to the list of preloads on line 962, you should be able to grab the `required` property directly from each assessment's linked questionnaire and skip using the resolver here entirely.